### ISSUE TITLE
Prometheus Operated Service w/ Multiple Prometheus CRs

### ISSUE THREAD

**Issue by michael-burt** [prometheus-operator/prometheus-operator#4647]
**TLDR**
- I have multiple Prometheus CRs in the same namespace
- Some of these include Thanos config stanza, others do not
- The `promethtues-operated` Service does not include the Thanos port mapping, even though it points to all of the Prometheus CRs I have deployed

**What did you do?**

I am running multiple Prometheus instances in the same namespace. This means that multiple Prometheuses are governed by the `prometheus-operated` service. Some of the CRs have Thanos enabled, and some don't. I am federating some high-cardinality metrics into their own Prometheus and only running Thanos Sidecar on the parent Promethtues.

My problem is that the `prometheus-operated` service is not including the Thanos sidecar port. Is there a way to force the `prometheus-operated` service to always target the Thanos sidecar port, even if some instances of operator-managed Prometheus do not include the Thanos configuration stanza?

https://github.com/prometheus-operator/prometheus-operator/blob/1ffeba6999f6c64345c34ee22f7ba544659f90c9/pkg/prometheus/statefulset.go#L315

**Did you expect to see some different?**

I am hoping there is a way I can make the `prometheus-operated` service include the Thanos port declaration.

**Environment**

Kubernetes 1.18.2
Prometheus Operator v0.55.0
Prometheus v2.33.4
Thanos v0.22.0
